extensions/amp-story/0.1/amp-story.js

```diff
@@ -936,6 +938,10 @@ export class AmpStory extends AMP.BaseElement {
    * @private
    */
   setHistoryStatePageId_(pageId) {
+    // Never save ad pages to history as they are unique to each visit.
+    if (pageId.slice(0, AD_PAGE_ID_PREFIX.length) === AD_PAGE_ID_PREFIX) {
```

Comment:
Hmm... Can we instead get page by id and check `isAd`?

Reply:
Sure, its slower though. Not sure if it matters

Reply:
If we get by id using the story function (not querying the DOM) I wouldn't expect it to be slow enough to matter, but I'm not in love with hardcoding the assumption of the ads page ids here

Reply:
Fair enough, I will change it. However we are in control of the creation of this id, and are guaranteeing it in [other places](https://github.com/ampproject/amphtml/blob/master/extensions/amp-story/amp-story-ads.md#tracking)

Reply:
If we just need this for tracking, maybe we should expose an analytics variable for whether the page is an ad?  It just feels like id is not very semantic